crates/runtime/src/datafusion/extension/bytes_processed.rs

```diff
@@ -285,27 +333,106 @@ impl ExecutionPlan for BytesProcessedExec {
             )
         };
 
-        let bytes_processed_stream = stream! {
-            let mut bytes_processed = 0u64;
-            while let Some(batch) = stream.next().await {
-                match batch {
-                    Ok(batch) => {
-                        bytes_processed += batch.get_array_memory_size() as u64;
-                        yield Ok(batch)
-                    }
-                    Err(e) => {
-                        yield Err(e);
-                    }
-                }
-            }
-            crate::metrics::telemetry::track_bytes_processed(
-                bytes_processed,
-                &request_context.to_dimensions(),
-            );
-        };
+        let bytes_processed_stream = BytesProcessedStream::new(stream, request_context);
 
         let stream_adapter = RecordBatchStreamAdapter::new(schema, bytes_processed_stream);
 
         Ok(Box::pin(stream_adapter))
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::datafusion::extension::bytes_processed::BytesProcessedExec;
+    use crate::{Runtime, RuntimeBuilder};
+    use arrow::array::Int64Array;
+    use arrow::record_batch::RecordBatch;
+    use datafusion::arrow::datatypes::{DataType, Field, Schema};
+    use datafusion::catalog::MemTable;
+    use datafusion::catalog::TableProvider;
+    use datafusion::common::Result;
+    use datafusion::physical_expr::expressions::col as physical_col;
+    use datafusion::physical_expr::{LexOrdering, PhysicalSortExpr};
+    use datafusion::physical_optimizer::optimizer::PhysicalOptimizer;
+    use datafusion::physical_plan::sorts::sort::SortExec;
+    use datafusion::physical_plan::{ExecutionPlan, displayable};
+    use std::sync::Arc;
+
+    fn make_test_table() -> Result<Arc<dyn TableProvider>> {
+        let schema = Arc::new(Schema::new(vec![Field::new("id", DataType::Int64, false)]));
+        let batch = RecordBatch::try_new(
+            Arc::clone(&schema),
+            vec![Arc::new(Int64Array::from_iter_values(0i64..10000))],
+        )?;
+
+        Ok(Arc::new(MemTable::try_new(schema, vec![vec![batch]])?))
+    }
+
+    #[allow(clippy::similar_names)]
+    #[allow(clippy::unwrap_used)]
```

Comment:
```suggestion
```

I'd prefer to avoid using unwrap in the codebase, even for tests and to prefer `.expect()`